provision.py

```diff
@@ -168,7 +170,8 @@ def main():
         run(["tools/setup/install-phantomjs", "--travis"])
     else:
         run(["tools/setup/install-phantomjs"])
-    run(["tools/setup/download-zxcvbn"])
+    if not os.path.exists("static/third/zxcvbn"):
+        run(["tools/setup/download-zxcvbn"])
```

Comment:
The other hunks look good, but I think this one will break updating the version of zxcvbn when we next want to do that.  Maybe makes sense to move this logic into `download-zxcvbn`, and have it check whether the sha1sum of zxcvbn.js is right inside that script?